### ISSUE TITLE
FieldBook crashes on launch

### ISSUE THREAD

**Issue by nia-h** [PhenoApps/Field-Book#10]
Hi -

I've been using FieldBook on a Lenovo Tab3 A8 all summer with no issues, but on updating to 4.0.1, it threw the error "Unique identifier has duplicates" (even though it doesn't) on attempting to import a field I'd been using. Now that it's on 4.0.2, it won't even launch - about a second after opening, I get the dialog "Unfortunately, FieldBook has stopped". Restarts and clearing caches have done nothing. I'm not sure if this is an issue with my tablet (as FieldBook works fine on my phone, although unfortunately all the data is on the tablet!) or something to do with the app.

Any advice would be greatly appreciated - thanks!
Nia

Edit to add - I've been working with two files for this project, each for a location trial. I've tried importing both frames to the app on my phone (also on 4.0.2) - just as on the tablet, one field works, while the other throws a "unique identifier contains duplicates" error.

Edit 2 - On another user profile on the tablet, FieldBook launches fine (although this means I don't have access to previously collected data). I have determined that the duplicates error seems to be because both fields have the same plot numbering (i.e. 1001-1240) and that's set as the unique ID. I can import either csv fine, provided it's the first to be imported, but the second will always throw the error.

**Comment by trife** [comment]
Hi Nia,
Can you upload the backup.db and backup.db_sharedpref.xml files here so I can attempt to recreate the error? I changed the database schema in v4 to more-heavily enforce the concept and requirements for unique identifiers- so that may be why you're running into issues.

**Comment by nia-h** [comment]
[NHughes_FieldBook_backups.zip](https://github.com/PhenoApps/Field-Book/files/1293695/NHughes_FieldBook_backups.zip)

Here you are - hope this helps.

**Comment by trife** [comment]
I'm able to import this without any issues. Rename both files on the tablet (e.g. nia.db and nia.db_sharedpref.xml), clear the data from the app in the app settings, and reimport the database and you should be good to go. Let me know if that works for you.

**Comment by nia-h** [comment]
FieldBook opens now, but all my trait data seems to be missing - I'm not sure if I'm missing a step to importing the database back into the app, or if that info is just gone? I have backups, but they're messier than I'd like to admit. Additionally, I still get the "duplicate" error when trying to have two spreadsheets imported at the same time, and only one particular spreadsheet will upload at all - the others give a "Error importing" message, even if there's nothing else imported yet.

If I can help troubleshoot this at all, please let me know.

**Comment by trife** [comment]
Sorry you're still having issues. Go ahead and send me an email and we can get this sorted.

**Comment by KariHugie** [comment]
Hi Trevor, 

My supervisor and I are having similar issues with the FieldBook app on our tablets. The app would not launch on his tablet after updating to 4.0.2, prompting the dialog "Unfortunately, FieldBook has stopped". However, I am not aware of any error messages indicating duplicates in the unique IDs. 

As per your previous post, I uploaded the backup.db files into the app on my phone. Initially it opened displaying the plot data. However, when I try to access data from other imported fields the app crashes and will no longer launch. 

I have attached the renamed .db files below and would greatly appreciate any advice you might have on this issue. 

[tc_backup.zip](https://github.com/PhenoApps/Field-Book/files/1318908/tc_backup.zip)

Thanks in advance.
